libobs/util/pipe-posix.c

```diff
@@ -93,5 +93,12 @@ size_t os_process_pipe_write(os_process_pipe_t *pp, const uint8_t *data,
 		return 0;
 	}
 
-	return fwrite(data, 1, len, pp->file);
+	size_t written = 0;
+	while (written < len) {
```

Comment:
Im not sure if anyone else has style suggestions. But the len check is unnecessary (fwrite with nmemb or size 0 returns 0). So this could just be written as a simple while loop (forgive me if theres a typo or 20).

```c
size_t wrote, written = 0;
while( (wrote = fwrite(...)) != 0 ) { written += wrote }
return written;
```

Also imo, `ret` should be used for the returned value of the surrounding value. Not used as a general variable.

Reply:
Using these variable names, I'm reminded of The Phantom of the Opera:

"And what is it that I'm meant to have wrote?"
(beat)
"Written."

Happy to do it either way.

Reply:
Yea mine is also not perfect, i leave it up to you.